web3_proxy/src/rpcs/request.rs

```diff
@@ -219,45 +225,48 @@ impl OpenRequestHandle {
         if let Err(err) = &response {
             // only save reverts for some types of calls
             // TODO: do something special for eth_sendRawTransaction too
-            let revert_handler = if let RequestRevertHandler::Save = revert_handler {
+            let error_handler = if let RequestErrorHandler::SaveRevert = revert_handler {
                 // TODO: should all these be Trace or Debug or a mix?
                 if !["eth_call", "eth_estimateGas"].contains(&method) {
                     // trace!(%method, "skipping save on revert");
-                    RequestRevertHandler::TraceLevel
+                    RequestErrorHandler::TraceLevel
                 } else if self.authorization.db_conn.is_some() {
                     let log_revert_chance = self.authorization.checks.log_revert_chance;
 
                     if log_revert_chance == 0.0 {
                         // trace!(%method, "no chance. skipping save on revert");
-                        RequestRevertHandler::TraceLevel
+                        RequestErrorHandler::TraceLevel
                     } else if log_revert_chance == 1.0 {
                         // trace!(%method, "gaurenteed chance. SAVING on revert");
                         revert_handler
                     } else if thread_fast_rng::thread_fast_rng().gen_range(0.0f64..=1.0)
                         < log_revert_chance
                     {
                         // trace!(%method, "missed chance. skipping save on revert");
-                        RequestRevertHandler::TraceLevel
+                        RequestErrorHandler::TraceLevel
                     } else {
                         // trace!("Saving on revert");
                         // TODO: is always logging at debug level fine?
                         revert_handler
                     }
                 } else {
                     // trace!(%method, "no database. skipping save on revert");
-                    RequestRevertHandler::TraceLevel
+                    RequestErrorHandler::TraceLevel
                 }
             } else {
                 revert_handler
             };
 
-            enum ResponseTypes {
+            // TODO: simple enum -> string derive?
+            #[derive(Debug)]
+            enum ResponseErrorType {
                 Revert,
                 RateLimit,
-                Ok,
+                Error,
```

Comment:
rename this back to `ResponseType` because this needs to have both "error" and "Ok" on it

Reply:
The whole enum is a ResponseType, no?

Reply:
or in other words, this enum is supposed to capture errors only?

Reply:
i'm fine with either, let me know what you prefer? this again (similar to the above one), occurs because stats_v2 and main simultaneously (and separate) modified this file. I will check if I can fix this